Providers/StaminaProviderConfig.jsx

```diff
@@ -0,0 +1,119 @@
+// /* eslint-disable react/destructuring-assignment */
+import React, { Component, createContext } from "react";
+import { AsyncStorage } from "react-native";
+import axios from "react-native-axios";
+
+const { Provider, Consumer } = createContext();
+
+class StaminaProvider extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      stamina: null,
+      date: null
+    };
+  }
+
+  async componentDidMount() {
+    const storedResult = await this.loadFromStorage();
+    try {
+      const resp = await axios.get("http://worldtimeapi.org/api/ip");
+
+      const date = new Date(resp.data.datetime);
+      const storedDate = new Date(storedResult);
+
+      if (this.state.date) {
+        if ((date - storedDate) / 1000 > 86400) {
+          AsyncStorage.setItem("date", JSON.stringify(date));
+          this.increaseStamina();
+        }
+      } else {
+        this.setState({ date: resp.data.datetime });
+        AsyncStorage.setItem("date", JSON.stringify(date));
+      }
+    } catch (err) {
+      console.log(err);
+    }
+  }
+
+  loadFromStorage = async () => {
+    const storedStamina = await this.getStoredStamina();
+    const storedDate = await this.getStoredDate();
+    if (!storedStamina) {
+      AsyncStorage.setItem("stamina", JSON.stringify(100));
+    }
+    this.setState({
+      stamina: storedStamina === null ? 100 : storedStamina,
```

Comment:
now we get some problems related to how we set async and state separately. Here if storedStamina were to be undefined, you would set async to 100, but state to undefined (2 different comparisions, `!` and `=== null`